pkg/install/resources.go

```diff
@@ -0,0 +1,161 @@
+package install
+
+import (
+	"fmt"
+
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Option func(*config)
+
+type config struct {
+	container string
+	tag       string
+}
+
+func (c *config) image() string {
+	return fmt.Sprintf("%s:%s", c.container, c.tag)
+}
+
+func WithContainer(container string) Option {
```

Comment:
Let's call this `WithImage`

Reply:
"image" usually implies a container and tag concatenated with a `:`. If I add `WithImage`, I would assume it should have different semantics (ie: specifying the whole image and not just the container)

Reply:
A container is the running entity that used the image for its root filesystem.

An image can either contain a tag or not. If the tag is omitted, it defaults to latest.

Reply:
ok, that makes sense. i decided to leave `WithImage` as the only functional option. The deployment will default to a pull-always pull policy and if a non-latest tag is specified, the deployment will be created with a pull-if-not-available pull policy.